### PR TITLE
chore: downgrade jest types

### PR DESCRIPTION
**Pull request by pauldambra**
we have jest types v29 and jest v27

this led me down a path of confusion where the types suggested you could do something you couldn't really

trying to upgrade to jest 29 caused _one of those_ cascading error -> upgrade -> sadness cycles

let's downgrade the types package instead

### CODE REVIEW

File: functional_tests/feature-flags.test.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
-/* eslint-disable compat/compat */
-import { v4 } from 'uuid'
```
Comment (pauldambra) [inline functional_tests/feature-flags.test.ts:2]: uuid package is incompatible in some way with jest 28+ and we don't need it, we only use it in this file, so we can uuidv7 instead

File: functional_tests/feature-flags.test.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+    // eslint-disable-next-line compat/compat
+    await new Promise<void>((resolve: () => void) => setTimeout(resolve, 500))
+}
 
 describe('FunctionalTests / Feature Flags', () => {
+    let token: string
+
+    beforeEach(() => {
+        token = uuidv7()
+    })
```
Comment (pauldambra) [inline functional_tests/feature-flags.test.ts:17]: a little tidying while we're in here

File: functional_tests/identify.test.ts [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
-        const anonymousId = posthog.get_distinct_id()
+    let token: string
+    let posthog: PostHog
+    let anonymousId: string
+
+    beforeEach(async () => {
+        token = uuidv7()
+        posthog = await createPosthogInstance(token)
+        anonymousId = posthog.get_distinct_id()
+    })
```
Comment (pauldambra) [inline functional_tests/identify.test.ts:19]: same same, a little tidying since we're removing uuid from here

File: package.json [hunk 4]
```diff
@@ -58,18 +58,16 @@
         "@rrweb/types": "2.0.0-alpha.13",
         "@sentry/types": "8.7.0",
         "@testing-library/dom": "^9.3.0",
-        "@testing-library/jest-dom": "^6.4.5",
+        "@testing-library/jest-dom": "^6.5.0",
         "@testing-library/preact": "^3.2.4",
         "@types/eslint": "^8.44.6",
-        "@types/jest": "^29.5.1",
+        "@types/jest": "^27.5.1",
```
Comment (pauldambra) [inline package.json:64]: the fix we came here to make

File: package.json [hunk 5]
```diff
@@ -58,18 +58,16 @@
...
         "@types/node": "^22.5.0",
         "@types/react-dom": "^18.0.10",
         "@types/sinon": "^17.0.1",
-        "@types/uuid": "^9.0.1",
         "@types/web": "^0.0.154",
         "@typescript-eslint/eslint-plugin": "^8.2.0",
         "@typescript-eslint/parser": "^8.2.0",
-        "babel-eslint": "10.1.0",
```
Comment (pauldambra) [inline package.json:72]: clearly not using this

File: package.json [hunk 6]
```diff
@@ -58,18 +58,16 @@
...
-        "@testing-library/jest-dom": "^6.4.5",
+        "@testing-library/jest-dom": "^6.5.0",
         "@testing-library/preact": "^3.2.4",
         "@types/eslint": "^8.44.6",
-        "@types/jest": "^29.5.1",
+        "@types/jest": "^27.5.1",
         "@types/node": "^22.5.0",
         "@types/react-dom": "^18.0.10",
         "@types/sinon": "^17.0.1",
-        "@types/uuid": "^9.0.1",
```
Comment (pauldambra) [inline package.json:68]: was weird that we were importing a function from a types package anyway

**Review by oliverb123 — APPROVED** [review]

**Pull request merged**